### PR TITLE
Enhancement/40 add outlook location field

### LINKED ISSUE #40: Implement the ability to add text in the Outlook "Location" field
### Contact Details

@kenneth-edwall

### What benefits does the suggestion solve?

This is a common feature in other meeting services that provides an outlook plugin. The "Location" field is a required field prior to sending a calendar invite. So if the field is blank Outlook will ask you whether you want to send the invite or not with a blank location field.

Instead of requering the end user put the information there manually the plugin could do it for them. E.g "Jitsi meeting" or "My meeting service"

### Feature suggestion description

Add logic in the plugin to fetch a variable that populates the Location field. Make it so that the variable can be configured by the administrator, preferably via e.g an xml or json file that the plugin reads.

The solution must accept åäö characters

### Alternative solutions

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Philldomd**
# Pull Request Description

Sets the Location field to a variable that is configured in the config.js file.
This will remove the question on send that blocks, asking if the user wants to send a message without location

Fixes #40 

## Checklist

- [x] Changes are limited to a single goal (avoid scope creep)
- [x] I confirm that I have read any Contribution and Development guidelines (CONTRIBUTING and DEVELOPMENT) and are following their suggestions.
- [x] I confirm that I wrote and/or have the right to submit the contents of my Pull Request, by agreeing to the _Developer Certificate of Origin_, (adding a 'sign-off' to my commits).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,17 +30,21 @@
     "@fluentui/react": "^8.52.3",
     "core-js": "^3.9.1",
     "es6-promise": "^4.2.8",
+    "eslint": "^7.32.0",
```
Comment (Ayko1595) [inline package.json:33]: I don't think we need to install eslint. Office introduces eslint and prettier through its boilerplate code

**Review by konstantin-ay — CHANGES_REQUESTED** [review]
Thanks again for your contribution, the feature works as expected! 😁 

But I have added some comments that I think should be adressed or discussed. Also, I was not able to run the tests locally. They seem to fail for the OfficeCallHandler.test.ts file

File: .gitignore
[the remark this answers is not in the document]
Comment (Philldomd) [reply .gitignore:1]: Yes, that is a file generated by npm install.

File: src/models/Config.ts [hunk 2]
```diff
@@ -7,6 +7,7 @@ export const defaultFontFamily = "Arial";
 
 interface Config {
   baseUrl?: string;
+  locationString?: string;
```
[the remark this answers is not in the document]
Comment (Philldomd) [reply src/models/Config.ts:10]: Sure I missed that

File: jest.config.js
[the remark this answers is not in the document]
Comment (Philldomd) [reply jest.config.js:1]: Will add

File: __tests__/OfficeCallHandler.test.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+//
+// SPDX-License-Identifier: MIT
+
+/**
+ * @jest-environment jsdom
+ */
+import * as configData from "../config.json";
+import Config from "../src/models/Config"
+const AddinOfficeMockObject = require("office-addin-mock");
+import { setLocationTest, setDataTest, addMeeting } from '../src/utils/OfficeCallHandler'
```
[the remark this answers is not in the document]
Comment (Philldomd) [reply __tests__/OfficeCallHandler.test.ts:11]: That is true :)

File: package-lock.json
[the remark this answers is not in the document]
Comment (Philldomd) [reply package-lock.json:1]: I can add it back for that purpose. But it had a lot of changes

File: package.json (hunk 1 above)
Comment (Philldomd) [reply package.json:33]: I guess it depends on IDE, the people that developed this parts used jetbrains ide.

File: LICENSES/CC0-1.0.txt
[the remark this answers is not in the document]
Comment (Philldomd) [reply LICENSES/CC0-1.0.txt:1]: They are changed when downloaded to windows. I missed to remove this commit here, I will push a new branch without this permission change

File: package.json (hunk 1 above)
Comment (Philldomd) [reply package.json:33]: I can remove and try without these npm packages.

File: tsconfig.json [hunk 4]
```diff
@@ -5,7 +5,7 @@
     "esModuleInterop": true,
     "experimentalDecorators": true,
     "jsx": "react",
-    "module": "CommonJS",
```
[the remark this answers is not in the document]
Comment (Philldomd) [reply tsconfig.json:8]: Hmm, no I think this snuck in here when I was looking for a bug

**Comment by Philldomd** [comment]
> Thanks again for your contribution, the feature works as expected! 😁
> 
> But I have added some comments that I think should be adressed or discussed. Also, I was not able to run the tests locally. They seem to fail for the OfficeCallHandler.test.ts file

What problem did you get here? Was npm install run to get latest changes in package?

File: LICENSES/CC0-1.0.txt
[the remark this answers is not in the document]
Comment (Ayko1595) [reply LICENSES/CC0-1.0.txt:1]: Couldn't you just fix them as part of a commit on the same branch?

File: .gitignore
[the remark this answers is not in the document]
Comment (Ayko1595) [reply .gitignore:1]: Yes I know that it is generated with npm install, but it is supposed to be checked in to the repository, as recommended by npm here: https://docs.npmjs.com/cli/v10/configuring-npm/package-lock-json 😊

File: package-lock.json
[the remark this answers is not in the document]
Comment (Ayko1595) [reply package-lock.json:1]: The package-lock.json will get a lot of changes whenever we add new dependencies, so it it expected 😊

**Comment by Ayko1595** [comment]
@Philldomd Yes I have the latest version of the code and also ran npm install. Here is the error message I received:

Screenshot 2024-06-11 at 13 13 14

File: LICENSES/CC0-1.0.txt
[the remark this answers is not in the document]
Comment (Philldomd) [reply LICENSES/CC0-1.0.txt:1]: I left it in its own commit so that i could preserve it on upload here. That commit  should never have been pushed.

**Comment by Philldomd** [comment]
@Ayko1595 If you change to the following will that work?
10 import { OfficeMockObject } from "office-addin-mock";
35 const setLocationMock = new OfficeMockObject(mockSetLocationServer) as any;
69 const setDataMock = new OfficeMockObject(mockSetDataServer) as any;

does the test work then?

**Comment by Ayko1595** [comment]
@Philldomd It does not work after those changes, I am still getting those errors. I think the issue could be the testing environment. The jest configuration specifies a jsdom environment, which is used on client-side testing. The tests you want to run in the OfficeCallHandler relies on a node environment which causes the test to fail. There is probably a mismatch in the configurations and the test files

**Comment by Philldomd** [comment]
I changed to Node instead and all other tests failed. I get everything to pass, both on my workstation and my laptop. So i can't reproduce this. I have removed package-lock and node_modules and run install again with the same resault. Have you added office-addin-mock?

**Comment by Ayko1595** [comment]
Office-addin-mock is installed yes. I am running this on MacOS and it still fails unfortunately. If you fix the comment regarding the Jest config's license header we will be able to run the GitHub action that tests the code

File: package.json (hunk 1 above)
Comment (Ayko1595) [reply package.json:33]: Did you try removing Eslint and tested without it?

Comment (Philldomd) [reply package.json:33]: I think i need to remove all changes in package.json and build towards a working version for node 14. I did not think 14 was still used here.

File: .github/workflows/workflow.yml [hunk 5]
```diff
@@ -17,4 +17,4 @@ jobs:
     uses: ./.github/workflows/license.yml
   test:
     needs: [license]
-    uses: ./.github/workflows/tests.yml
```
Comment (Ayko1595) [inline .github/workflows/workflow.yml:20]: Hmm I think GitHub is drunk? I can't see any change here

**Comment by janderssonse** [comment]
Off-topic: Nice to see you at the meeting @Philldomd and while have 0% time left this week to help out reviewing, I'm very glad to see these PR's arrive to the project, and that FK is part of this. Much cred to all involved in this. This is collaborationn for real.

**Comment by Philldomd** [comment]
Code is squashed and updated as requested

**Comment by Philldomd** [comment]
Testing with older package.json file

File: package.json [hunk 6]
```diff
@@ -81,10 +83,23 @@
     "tabWidth": 2,
     "printWidth": 200
   },
+  "eslintConfig": {
```
Comment (Ayko1595) [inline package.json:86]: Just a minor thing: 
Do we need to have this eslintConfig in the package.json anymore?

Comment (Philldomd) [reply package.json]: I think that was added for an IDE plugin to work with eslint. Not sure, removed it for now

**Review by konstantin-ay — APPROVED** [review]
Looks good! 😄🤝

**Comment by Philldomd** [comment]
I found out now from a co-worker that this actually introduces a bug, the location field is cleared from other information upon generating a jitsi link. I will have a look into i and ensure that this does not happen here. Await my next commit for that fix and a testcase that checks it.

**Comment by Ayko1595** [comment]
@Philldomd Oh okay, well nice catch by our co-worker. If it is possible to introduce a unit test that tests this issue it would be great :-)

**Review by konstantin-ay — CHANGES_REQUESTED** [review]
Bug was found so I change this status to "Request changes"

**Comment by Philldomd** [comment]
Okey now I think this one is finally finished. Added a lot of tests to control the location field. At least it is not duplicating the text or removing other location field information.

**Comment by Ayko1595** [comment]
Looks good @Philldomd but there seems to be a conflict in the commands.ts file

**Comment by Philldomd** [comment]
Let's wait for bugfix and then merge this code in. There will be some major changes when that happens

**Review by konstantin-ay — APPROVED** [review]

**Pull request merged**